rpc-client-api/Cargo.toml

```diff
@@ -19,6 +19,7 @@ serde = "1.0.144"
 serde_derive = "1.0.103"
 serde_json = "1.0.83"
 solana-account-decoder = { path = "../account-decoder", version = "=1.15.0" }
+solana-ledger = { path = "../ledger", version = "=1.15.0" }
```

Comment:
Another alternative is actually to remove the enum altogether, and update `Blockstore::get_recent_perf_samples()` return type to `Result<Vec<(Slot, PerfSampleV2)>>`, and then revert the rpc.rs changes. But I'm guessing that won't appeal to you.

Reply:
> Another alternative is actually to remove the enum altogether, and update Blockstore::get_recent_perf_samples() return type to Result<Vec<(Slot, PerfSampleV2)>>, and then revert the rpc.rs changes. But I'm guessing that won't appeal to you.

`blockstore` may contain either `PerfSampleV1` or `PerfSampleV2`.
Are you are saying that in this particular case, we could just set `non_vote_transaction_count` to `0` when reading an older stat?
In this case, the client side would not be able to tell a difference between a case when a block is missing the stat from the case when it has no non-vote transactions.

Reply:
> Hmm, I'd prefer not to add this dependency to solana-rpc-client-api, if we can help it. We've been doing a lot of work to minimize the dependencies in client crates so that downstream users can limit their imports.

It has been some time since I worked with Rust dependencies.
I thought that package clients do not need to include any indirect dependencies, do they?
Or are they still pulled in by `cargo`?

> Possible alternatives: implement the PerfSample-to-RpcPerfSample conversion in solana-rpc, or implement methods like num_transactions(), num_slots(), etc on PerfSample to call in rpc.rs

Moved conversion into `rpc_perf_sample_from_perf_sample` inside `solana_rpc`.
(An orphan instance would have been a nice option here.)